docs/source/query/overview.rst

```diff
@@ -5,6 +5,57 @@ Querying Snuba
 This guide drives you through the process of authoring and testing a
 Snuba query.
 
+
+Understanding the cost of queries
+=================================
+The cost of a snuba query has many factors but these are key to keep in mind when constructing queries:
+
+* Amount of rows scanned
+* Amount of columns scanned
+* Type of data scanned
```

Comment:
I think this is a bit of an oversimplification. Do you plan to expand this section or do you plan to keep this level of detail? (I think we should expand and provide a lot more details over time. Sadly when addressing the performance of clickhouse queries there are very few areas where we can have simple absolute rules, for all practical purposes it is critical to understand what clickhouse would do in the specific case).

On sharded datasets the type of aggregtions that happen on the distributed table are equally important because those do not parallelize well.

The type of conditions to run are also critical: equality is one thing, regex is very different.

Reply:
You may want to be aware of this problem before going with that strategy.
Out of experience explaining query performance to product teams, one constant I saw is that, if we provide a simple rule ignoring the corner cases, the product team takes it literally. The result is that they will derive their strategy from that simple rule and more often than not they will run into corner case where the rule does not apply.
Example: 2 years ago the message was "conditions on tags are bad". That became the universal rule and we still did not manage to get the nuance around "which conditions on tags are bad and which ones are ok" through.

If I were you I would try to cover a smaller area but providing details for it. 
Just as an example: You could focus on rows scanned first. And provide a longer list of ways to reduce the rows scanned. Then in a follow up talk about reducing columns scanned with its details. 
But, your call.